book/src/getting-started/2-our-first-cxx-qt-module.md

```diff
@@ -31,49 +32,57 @@ This is a lot to take in, so let's go one step at a time.
 Starting with the module definition:
 ```rust,ignore
 {{#include ../../../examples/qml_minimal/rust/src/cxxqt_object.rs:book_bridge_macro}}
+    // ...
+}
 ```
 
-Because we add the `#[cxx_qt::bridge]` macro to the module definition,
-CXX-Qt will look inside the module for further macros which can define the QObject.
+A `#[cxx_qt::bridge]` is the same as a `#[cxx::bridge]` and you can use all features of CXX in it.
+Additionally, a `#[cxx_qt::bridge]` gives you a few more features that allow you to create QObjects.
 
-For the `#[cxx_qt::bridge]` macro to work, we first need to define the properties that will be exposed in the new C++ object.
-This is done by tagging fields with `#[qproperty]`:
+To create a new QObject subclass, we can define a struct within our module and mark it with `#[cxx_qt::qobject]`.
+Additionally, we need to either `impl Default` or `#[derive(Default)]` for our struct.
 ```rust,ignore
 {{#include ../../../examples/qml_minimal/rust/src/cxxqt_object.rs:book_rustobj_struct}}
 ```
-That means the newly created QObject subclass will have two properties as members: `number` and `string`. For names that contain multiple words, like `my_number`, CXX-Qt will perform the snake_case to camelCase conversion to fit with C++/QML naming conventions.
+The Rust struct can be defined just like a normal Rust struct and can contain any kind of field, even Rust-only types.
+If a field is marked as `#[qproperty]` it will be exposed to the C++ side as a `Q_PROPERTY`.
 
-Note that the data types we use here are normal Rust data types.
-CXX-Qt will automatically convert these types to their C++/Qt equivalent.
+That means the newly created QObject subclass will have two properties as members: `number` and `string`. For names that contain multiple words, like `my_number`, CXX-Qt will automatically rename the field from snake_case to camelCase to fit with C++/QML naming conventions (e.g. `myNumber`).
+
+Do note though that any fields marked as `#[qproperty]` must be types that CXX can translate to C++ types.
 In our case that means:
 - `number: i32` -> `int number`
```

Comment:
This should be `qint32` not `int` (and ideally `::std::int32_t` but IIRC Qt 5 had issues)

Reply:
Done